common/djangoapps/entitlements/models.py

```diff
@@ -59,8 +63,8 @@ def is_entitlement_regainable(self, entitlement):
 
                     # This is < because now - some duration being equal to regain_period_days means that
                     # many days have passed, which should then make the entitlement no longer regainable
-                    (now - course_overview.start).days < self.regain_period_days or
-                    (now - entitlement.enrollment_course_run.created).days < self.regain_period_days) and
+                    ((now - course_overview.start).days < self.regain_period_days or
+                     (now - entitlement.enrollment_course_run.created).days < self.regain_period_days)) and
                 # This is <= because a days_until_expiration 0 means that the expiration day has not fully passed yet
                 # and that the entitlement should not be expired as there is still time
                 self.get_days_until_expiration(entitlement) >= 0
```

Comment:
I think if `get_days_until_expiration` takes into account the regain period logic, this should be the only thing that needs to be checked when determining if the entitlement is regainable.